src/pages/EditClientPage.tsx

```diff
@@ -0,0 +1,175 @@
+// import { useRef } from "react";
+import { useUpdateClient } from "../api/mutations";
+import { useFormik } from "formik";
+import { clientValidationSchema } from "../validators/validators";
+import TextField from "@mui/material/TextField";
+import { useNavigate, useParams, Link } from "react-router-dom";
+import { useGetClientById } from "../api/queries";
+import { ROUTES } from "../routes";
+import { useEffect } from "react";
+
+type FormValues = {
+  imgSrc: string;
+  name: string;
+  surname: string;
+  street: string;
+  postCode: string;
+  town: string;
+  region: string;
+  phoneNumber: string;
+};
+
+export const EditClientPage = () => {
+  const params = useParams();
+  const clientId: string = params.id ?? "";
+  // ? Czy to powinno mieć jakiś mechanizm że jeśli ID puste to nie wykonuj całej reszty?
+  const navigate = useNavigate();
+  const { mutate } = useUpdateClient(clientId);
+  const { data, isError, isLoading } = useGetClientById(params.id);
+
+  const formik = useFormik<FormValues>({
+    initialValues: {
+      imgSrc: "",
+      name: "",
+      surname: "",
+      street: "",
+      postCode: "",
+      town: "",
+      region: "",
+      phoneNumber: "",
+    },
+    onSubmit: (values: FormValues) => {
+      mutate(values);
+      navigate(ROUTES.clients);
+    },
+    validationSchema: clientValidationSchema,
+  });
+
+  useEffect(() => {
+    if (data) {
+      formik.setValues({
+        imgSrc: data.imgSrc || "",
+        name: data.name || "",
+        surname: data.surname || "",
+        street: data.street || "",
+        postCode: data.postCode || "",
+        town: data.town || "",
+        region: data.region || "",
+        phoneNumber: data.phoneNumber || "",
+      });
+    }
+  }, [data]);
+  // ? Co mu tu nie pasuje?
```

Comment:
sprawdź czy jak przypiszesz te wartości do obiektu z typem FormValues to nadal tak będzie